enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/AutoTLSEnricher.java

```diff
@@ -0,0 +1,190 @@
+package io.fabric8.maven.enricher.fabric8;
+
+import io.fabric8.kubernetes.api.builder.TypedVisitor;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.PodSpecBuilder;
+import io.fabric8.kubernetes.api.model.PodTemplateSpecBuilder;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+import io.fabric8.maven.enricher.api.Kind;
+import io.fabric8.utils.Strings;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Enriches declarations with auto-TLS annotations, required secrets reference,
+ * mounted volumes and PEM to keystore converter init container.
+ *
+ * This is opt-in so should not be added to default enrichers as it only works for
+ * OpenShift.
+ */
+public class AutoTLSEnricher extends BaseEnricher {
+    static final String ENRICHER_NAME = "fmp-autotls";
+    static final String AUTOTLS_ANNOTATION_KEY = "service.alpha.openshift.io/serving-cert-secret-name";
+
+    private String secretName;
+
+    enum Config implements Configs.Key {
+        tlsSecretName,
+
+        tlsSecretVolumeMountPoint  {{ d = "/var/run/secrets/fabric8.io/tls-pem"; }},
+
+        tlsSecretVolumeName        {{ d = "tls-pem"; }},
+
+        jksVolumeMountPoint        {{ d = "/var/run/secrets/fabric8.io/tls-jks"; }},
+
+        jksVolumeName              {{ d = "tls-jks"; }},
+
+        pemToJKSInitContainerImage {{ d = "jimmidyson/pemtokeystore:v0.1.0"; }},
+
+        pemToJKSInitContainerName  {{ d = "tls-jks-converter"; }},
+
+        keystoreFileName           {{ d = "keystore.jks"; }},
+
+        keystorePassword           {{ d = "changeit"; }},
+
+        keystoreCertAlias          {{ d = "server"; }};
+
+        public String def() { return d; } protected String d;
+    }
+
+    public AutoTLSEnricher(EnricherContext buildContext) {
+        super(buildContext, ENRICHER_NAME);
+
+        this.secretName = getConfig(Config.tlsSecretName);
+        if (Strings.isNullOrBlank(secretName)) {
+            this.secretName = getProject().getArtifactId() + "-tls";
+        }
+    }
+
+    @Override
+    public Map<String, String> getAnnotations(Kind kind) {
+        if (!isOpenShiftMode()) {
+            return super.getAnnotations(kind);
```

Comment:
You should probably return `null` directly here since your know that your enricher doesn't want to add any annotations in this case.

Actually I would collapse this method in sth like

```java
if (!isOpenShiftMode() || !kind.isService()) {
   return null;
}
return Collections.singletonMap(AUTOTLS_ANNOTATION_KEY, this.secretName);
```